src/NServiceBus.Core/Persistence/Learning/SagaPersister/SagaStorageFile.cs

```diff
@@ -40,40 +38,52 @@ public void Dispose()
             fileStream = null;
         }
 
-        public static bool TryOpen(Guid sagaId, SagaManifest manifest, out SagaStorageFile sagaStorageFile)
+        public static Task<SagaStorageFile> Open(Guid sagaId, SagaManifest manifest)
         {
             var filePath = manifest.GetFilePath(sagaId);
 
             if (!File.Exists(filePath))
             {
-                sagaStorageFile = null;
-                return false;
+                return Task.FromResult<SagaStorageFile>(null);
             }
 
-            sagaStorageFile = new SagaStorageFile(new FileStream(filePath, FileMode.Open, FileAccess.ReadWrite, FileShare.ReadWrite, DefaultBufferSize, FileOptions.Asynchronous), manifest);
-
-            return true;
+            return OpenWithDelayOnConcurrency(manifest, filePath, FileMode.Open);
         }
 
-        public static SagaStorageFile Create(Guid sagaId, SagaManifest manifest)
+        public static Task<SagaStorageFile> Create(Guid sagaId, SagaManifest manifest)
         {
             var filePath = manifest.GetFilePath(sagaId);
 
-            return new SagaStorageFile(new FileStream(filePath, FileMode.CreateNew, FileAccess.ReadWrite, FileShare.ReadWrite, DefaultBufferSize, FileOptions.Asynchronous), manifest);
+            return OpenWithDelayOnConcurrency(manifest, filePath, FileMode.CreateNew);
         }
 
+        static async Task<SagaStorageFile> OpenWithDelayOnConcurrency(SagaManifest manifest, string filePath, FileMode fileAccess)
+        {
+            try
+            {
+                return new SagaStorageFile(new FileStream(filePath, fileAccess, FileAccess.ReadWrite, FileShare.None, DefaultBufferSize, FileOptions.Asynchronous), manifest);
+            }
+            catch (IOException)
+            {
+                // give the other task some time to complete the saga to avoid retrying to much
+                await Task.Delay(100)
+                    .ConfigureAwait(false);
+
+                return new SagaStorageFile(new FileStream(filePath, fileAccess, FileAccess.ReadWrite, FileShare.None, DefaultBufferSize, FileOptions.Asynchronous), manifest);
```

Comment:
Fair enough, since the other persisters lean on immediate retries as well it might be good to use the same pattern. Will change